### PR TITLE
Add non-code linters

### LINKED ISSUE #11: Project maintenance: Markdown linter
Add a linter for the `.md` files

### LINKED ISSUE #12: Project maintenance: Action linter
Add a linter for the `.yml` files (GitHub Actions -specific)

### LINKED ISSUE #20: Project maintenance: YAML linter
Add a linter for the `.yml` files

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
* Markdown
* YAML
* GitHub Actions

Closes #11.
Closes #12.
Closes #20.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    steps:
+      - uses: actions/checkout@v4.1.7
+
+      - name: markdownlint
+        uses: DavidAnson/markdownlint-cli2-action@v17.0.0
+        with:
+          globs: |
+            *.md
+          # .github/**/*.md
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/lint.yaml:27]: Not yet available, but maybe soonish 🤷

File: .github/workflows/build.yaml
Comment (paulo-ferraz-oliveira) [inline .github/workflows/build.yaml:1]: If we want we can be more relaxed on this, given the `.yaml` already existed.

File: .github/workflows/update_builds_txt.yaml
Comment (paulo-ferraz-oliveira) [inline .github/workflows/update_builds_txt.yaml:1]: If we want we can be more relaxed on this, given the `.yaml` already existed.

**Review by starbelly — APPROVED** [review]
LGTM

**Comment by paulo-ferraz-oliveira** [comment]
I'm merging. It should be easy to revert if this should errors in the default branch.

**Pull request merged**